### PR TITLE
[HttpFoundation][HttpKernel] Remove unused code and useless casts

### PR DESCRIPTION
**Pull request by alexandre-daubois**
| Q             | A
| ------------- | ---
| Branch?       | 7.2
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | -
| License       | MIT

Removed some superfluous code and useless casts across HttpFoundation and HttpKernel

### CODE REVIEW

**Review by xabbuh — APPROVED** [review]

File: src/Symfony/Component/HttpKernel/Kernel.php [hunk 1]
```diff
@@ -530,7 +530,7 @@ protected function initializeContainer(): void
 
         $buildDir = $this->container->getParameter('kernel.build_dir');
         $cacheDir = $this->container->getParameter('kernel.cache_dir');
-        $preload = $this instanceof WarmableInterface ? (array) $this->warmUp($cacheDir, $buildDir) : [];
+        $preload = $this instanceof WarmableInterface ? $this->warmUp($cacheDir, $buildDir) : [];
```
Comment (AlexisRossat) [inline src/Symfony/Component/HttpKernel/Kernel.php:533]: Is the array cast on line 542 also removable ? 
I'm not sure PHP while do it automatically.

Comment (alexandre-daubois) [reply src/Symfony/Component/HttpKernel/Kernel.php:533]: Indeed :+1:

**Comment by alexisLefebvre** [comment]
In another PR, it was suggested to keep explicit casts: https://github.com/symfony/symfony/pull/57758#discussion_r1682790234

Is there a guideline about this topic?

**Comment by xabbuh** [comment]
If I don't miss anything, the places where the explicit cast was removed does not change anything as the desired type is already present and removing the cast won't lead to an implicit type cast.

**Comment by alexandre-daubois** [comment]
That's right. I guess derrabus wanted to keep the explicit cast because it is done on a Stringable object. In this PR, the removed casts were useless and my guess is that those casts were introduced when the codebase wasn't as typed as it is now.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @alexandre-daubois.

**Pull request merged**